metering_billing/postgres_usage_ledger.py

```
@@ -88,97 +91,369 @@
+        with self._gate:
+            return self._closed
+
+    @staticmethod
+    def _close_quietly(connection: Any) -> None:
+        """Close one owned session, swallowing secondary close failures."""
+        try:
+            close = getattr(connection, "close", None)
+            if callable(close):
+                close()
+        except Exception:  # noqa: BLE001 - shutdown must not mask the original error
+            pass
```

Ruff (0.16.2)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)